drivers/mcp2515/candev_mcp2515.c

```diff
@@ -449,66 +451,67 @@ static int _set_filter(candev_t *dev, const struct can_filter *filter)
         f.can_mask &= CAN_SFF_MASK;
     }
 
-    /* Browse on each mailbox to find an empty space */
-    int mailbox_index = 0;
-
-    while (mailbox_index < MCP2515_RX_MAILBOXES) {
-        /* mask unused */
-        if (dev_mcp->masks[mailbox_index] == 0) {
-            /* set mask */
-            mcp2515_set_mask(dev_mcp, mailbox_index, f.can_mask);
-            /* set filter */
-            mcp2515_set_filter(dev_mcp, MCP2515_FILTERS_MB0 * mailbox_index,
-                               f.can_id);
+    /* mask unused */
+    if (dev_mcp->masks[f.target_mailbox] == 0) {
+        /* set mask */
+        mcp2515_set_mask(dev_mcp, f.target_mailbox, f.can_mask);
```

Comment:
benpicco's comment down below made me wonder... shouldn't there be a mutex_lock here too? 

(idem line 459)